elements/drag-drop-elements/pl-drag-drop-element.py

```diff
@@ -0,0 +1,389 @@
+
+
+def prepare(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    answerName = pl.get_string_attrib(element, 'answers-name')
+
+    mcq_options = []
+    correct_answers = []
+    correct_answers_indent = []
+
+    isShuffle = pl.get_string_attrib(element, 'shuffle-options', 'false')  # default to FALSE, no shuffling unless otherwise specified
+
+    for html_tags in element:
+        if html_tags.tag == 'pl-answer':
+            # CORRECT is optional for backward compatibility
+            # pl.check_attribs(html_tags, required_attribs=[''], optional_attribs=['correct', 'ranking', 'indent'])
+            mcq_options.append(str.strip(html_tags.text))   # store the original specified ordering of all the MCQ options
+        # BACKWARD COMPATIBILITY CODE
+        if html_tags.tag == 'pl-distractor':
+            mcq_options.append(str.strip(html_tags.text))   # store the original specified ordering of all the MCQ options
+
+    if isShuffle == 'true':
+        random.shuffle(mcq_options)
+
+    for html_tags in element:
+        if html_tags.tag == 'pl-answer':
+            isCorrect = pl.get_string_attrib(html_tags, 'correct', 'true')  # default to true, for backward compatibility
+            answerIndent = pl.get_string_attrib(html_tags, 'indent', '-1')  # get answer indent, and default to -1 (indent level ignored)
+            if isCorrect.lower() == 'true':
+                # add option to the correct answer array, along with the correct required indent
+                correct_answers.append(str.strip(html_tags.text))
+                correct_answers_indent.append(answerIndent)
+
+    data['params'][answerName] = mcq_options
+    data['correct_answers'][answerName] = {'correct_answers': correct_answers,
+                                           'correct_answers_indent': correct_answers_indent}
+
+
+def parse(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    answerName = pl.get_string_attrib(element, 'answers-name')
+
+    temp = answerName
+    temp += '-input'  # this is how the backend is written
+
+    student_answer_temp = ''
+    if temp in data['raw_submitted_answers']:
+        student_answer_temp = data['raw_submitted_answers'][temp]
+
+    if student_answer_temp is None:
+        data['format_errors'][answerName] = 'NULL was submitted as an answer!'
+        return
+    elif student_answer_temp == '':
+        data['format_errors'][answerName] = 'No answer was submitted.'
+        return
+
+    student_answer_temp = list(student_answer_temp.rsplit(',,,'))
+
+    student_answer = []
+    student_answer_indent = []
+    permutationMode = pl.get_string_attrib(element, 'permutation-mode', 'html-order')
+
+    student_answer_ranking = ['Question permutationMode is not "ranking"']
+    for answer in student_answer_temp:
+        # student answers are formatted as: {answerString}:::{indent}, we split the answer
+        answer = answer.rsplit(':::')
```

Comment:
As we were talking about in the comments, we should do something more elegant like storing the answers in a JSON object, not just separating them by some string

Reply:
yep, answers are now stored in an JSON object